client/adventureMap/CList.h

```diff
@@ -150,8 +150,9 @@ class CTownList	: public CList
 	class CTownItem : public CListItem
 	{
 		std::shared_ptr<CAnimImage> picture;
+		CTownList *parentList;
 	public:
-		const CGTownInstance * const town;
+		int townPos;
```

Comment:
it's not Pos (usually townPos in other places is town map position, like int3), but index.
The problem with indices is that is's possible to get out-of-range exception (and access violation in your code, as you use operator [] without checking index).

What's wrong with pointer? You can get current position of stored pointer indside array using vstd::find_pos and internally get next/prev town pointer. Less changes -> lower risk then

Reply:
The problem with pointer was: That a pointer wasn't valid anymore and I got segmentation fault. With the index the problem no longer exists.

Reply:
You can test build before `fix segmentation fault; update townlist on adventuremap`. Move towns in town window and then click onto it to change active town. Sometime it crashes. After the change everything is fine.

Reply:
then maybe use ObjectInstanceId? Or rename to townIndex?

Reply:
can rename it